Tests/UnitTests/Networking/Backend/BackendPostOfferForSigningTests.swift

```diff
@@ -106,27 +106,24 @@ class BackendPostOfferForSigningTests: BaseBackendTests {
         let offerIdentifier = "offerid"
         let discountData = "an awesome discount".data(using: String.Encoding.utf8)!
 
-        var receivedError: NSError?
-        var receivedUnderlyingError: NSError?
+        var receivedError: BackendError?
 
         backend.post(offerIdForSigning: offerIdentifier,
                      productIdentifier: productIdentifier,
                      subscriptionGroup: group,
                      receiptData: discountData,
                      appUserID: Self.userID) { result in
-            receivedError = result.error as NSError?
-            receivedUnderlyingError = receivedError?.userInfo[NSUnderlyingErrorKey] as? NSError
+            receivedError = result.error
         }
 
         expect(receivedError).toEventuallyNot(beNil())
-        expect(receivedError?.domain).toEventually(equal(RCPurchasesErrorCodeDomain))
-        expect(receivedError?.code).toEventually(
-            equal(ErrorCode.unexpectedBackendResponseError.rawValue))
-        expect(receivedUnderlyingError?.code).toEventually(
-            equal(UnexpectedBackendResponseSubErrorCode.postOfferIdMissingOffersInResponse.rawValue))
+
+        expect(receivedError) == .unexpectedBackendResponse(.postOfferIdMissingOffersInResponse)
```

Comment:
One line to compare the error instead of 5 🙃

Reply:
Yeah, this is reallllly nice